camel/agents/chat_agent.py

```diff
@@ -370,97 +373,73 @@ def step(
                 response_id,
             ) = self._step_model_response(openai_messages, num_tokens)
 
+            # If the model response is not a function call, meaning the model
+            # has generated a message response, break the loop
             if (
-                self.is_tools_added()
-                and isinstance(response, ChatCompletion)
-                and response.choices[0].message.tool_calls is not None
+                not self.is_tools_added()
+                or not isinstance(response, ChatCompletion)
+                or response.choices[0].message.tool_calls is None
```

Comment:
I think it's the desired behavior in current design. Acutally the tool_choice config should be managed by the agent rather than the user.

Reply:
I think it's the desired behavior in current design. Acutally the tool_choice config should be managed by the agent rather than the user.

Reply:
Hey @dandansamax , I think since currently we support `tool_choice` in LLM parameter then we need to make sure there will not have this kind of infinite loop in `ChatAgent`, I discussed with @WHALEEYE , we can handle `tool_choice` separately  when doing `ChatAgent` refactor further